### ISSUE TITLE
Syntax highlighting

### ISSUE THREAD

**Issue by letmaik** [JasonStein/Notepads#74]
It would be good if syntax highlighting was supported.

My current go-to notepad replacement is http://www.flos-freeware.ch/notepad2.html which is blazingly fast, has syntax highlighting, but is not maintained anymore.

**Comment by JasonStein** [comment]
I do not have plan to add syntax highlighting at this moment. I wanted to keep Notepads as Notepad, meaning everyone can use it to write notes, edit config on the fly but not to use it as a coding tool. You should probably use VS Code or Notepad++ in these kind of scenarios. Another reason I do not want to add syntax highlight is that it takes CPU resources and might end up increasing app launch time. It also makes it hard to deal with large files.

**Comment by letmaik** [comment]
Please look at http://www.flos-freeware.ch/notepad2.html. It can deal with large files and has no issues with syntax highlighting. For me it's a deal-breaker to be honest as syntax highlighting is standard by now even in simple editors.

**Comment by entdark** [comment]
> Please look at http://www.flos-freeware.ch/notepad2.html. It can deal with large files and has no issues with syntax highlighting. For me it's a deal-breaker to be honest as syntax highlighting is standard by now even in simple editors.

You are trying to crack nuts with a sledgehammer.
Use that notepad2 if it fits your needs, or any other advanced editor like Notepad++ or Sublime Text.

**Comment by DennstedtB** [comment]
I don't need Syntax highlightning directly in my text / config files ( which is the main reason for using such small and fast editors.. at least in my opinion ). 

What I like is the markdown integration and the syntax highlightning with the "```<lang>" tag - What I'd like to know is, is it somehow possible to add the highlightning  for some other languages ( e. g. I am developing mostly in abap and python .. I am documentating mostly everything with markdown files )? 
You are mentioning something like that in Screenshot 2 ... 

So is it possible to add own syntax highlightning for .md files? ( if yes, how? )

**Comment by AkazaRenn** [comment]
I would really like to see if there can be support for syntax highlighting for formats of config files like JSON and YAML, that's not to use Notepads as a programming tool but it would really help when using it to edit configs.

**Comment by MarcelCoding** [comment]
_My Feature request from #326_

## Syntax Highlightning

### Is your feature request related to a problem? Please describe.
It would be cool if configuration formats such as YAML or XML were provided with syntax highlighting.

### Describe the solution you'd like
Syntax highlighting for configuration formats. (YAML, XML, ...)

**Comment by luojunyuan** [comment]
About the performance maybe we can add this as a option feature. Not a coding tool but I think It would be nice to read

**Comment by Frioo** [comment]
I think syntax highlighting is crucial.
It would not make this project a code editor, but editing json, xml and other configuration files
is much easier with some basic highlighting!
Don't get me wrong but - even Discord (a voip/chat app) support syntax highlighting in messages!
It could be optional, of course, but please consider it if possible.

**Comment by P-de-Jong** [comment]
I agree with @Frioo and @AkazaRenn 
I don't use any notepad application for really editing code, but highlighting is very useful even when just editing a JSON or other files.

**Comment by entdark** [comment]
> Don't get me wrong but - even Discord (a voip/chat app) support syntax highlighting in messages!
> It could be optional, of course, but please consider it if possible.
Discord is basically a web-page that definitely use some predefined syntax highlighting js-framework. So it's a bad comparison.

**Comment by obskyr** [comment]
A lack of syntax highlighting is the only thing keeping me from using Notepads – it's pretty much perfect, *except* for the fact that editing XML and one-off bash scripts and such is impractical due to a lack of syntax highlighting!

**Comment by ahvth** [comment]
I am using this as my notepad replacement regardless, but if it had syntax highlighting it would also fully replace Notepad++ for me!

I love that you don't want this to be a replacement for VSCode, atom, sublime or whatever, that space is crowded enough, but syntax highlighting would still be nice! :)

**Comment by blachawk** [comment]
I agree too the comments on here AND with the stance of the owners of the project.  I would at least beg in mercy for JavaScript syntax highlighting that will complement the html and css highlighting already part of the project.  pretty pleeeeeaaassseee!!!!!

**Comment by gmr458** [comment]
Is a version with syntax highlighting and one without syntax highlighting possible?

**Comment by ultravelocity** [comment]
I have to agree with the others. Notepads would be the perfect replacement for me if it had syntax highlighting. Of course I use other more appropriate tools for development, but every once in awhile I open a ini, json, xml, sql, js, html, or even cs file (currently I use notepad2), and the highlighting is tremendously useful. Perhaps a fork should be considered if this isn't a priority.

For now I use Notepad2 with Stardock Groupy for tabs, but it's not very pretty!

**Comment by xerz-one** [comment]
I wonder what the right approach to implement this would be — perhaps with a fork of an existing library like https://github.com/thomasjo/highlight or https://github.com/elvirbrk/NoteHighlight2016 would suffice? Maybe with a C(++) library like [Highlight](https://gitlab.com/saalen/highlight)?

**Comment by livejamie** [comment]
Simple syntax highlighting like Marxico does would be amazing

![image](https://user-images.githubusercontent.com/49513/147820833-ad0ab010-a552-4a2b-93ce-6595a2568b88.png)

**Comment by kamik-hub** [comment]
So?
